projects/playground/src/app/app.component.ts

```diff
@@ -51,6 +59,13 @@ export class AppComponent {
     } else {
       this.auth.loginWithRedirect({
         ...(this.organization ? { organization: this.organization } : null),
+        appState: {
+          deeply: {
+            nested: {
+              value: this.loginOptionsForm.value.appState,
+            },
+          },
+        },
```

Comment:
Same here, it confuses me  that we are using `appState.deeply.nested.value = this.loginOptionsForm.value.appState`, it should either be `appState: appState` or `appState.deeply.nested.value = this.loginOptionsForm.value.someValue`.

However, I said before I would prefer to reduce the nesting.